### PR TITLE
Remove unnecessary required annotation

### PR DESCRIPTION
**Pull request by gfxcc**
Many fields are marked as `required` unnecessarily. That introduce several issues
* User needs to specified fields they do not care.  E.g. for `CaPool`, `baseKeyUsage` must also be specified when user only want to specify `extendedKeyUsage`.
* User are not able to create a certificate with only `SAN` be specified since `Subject` is marked as required.
* Examples become too large since non-related fields have to be specified.

b/293361953

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note: note
privateca: remove `required` annotation from several fields
```

### CODE REVIEW

**Comment by roaks3** [comment]
The change looks good, but it would be nice to verify that these changes are all correct. Could you add an example (or modify an existing one) that omits as many fields as possible?

File: mmv1/products/privateca/CaPool.yaml [hunk 1]
```diff
@@ -130,8 +130,7 @@ properties:
                 - !ruby/object:Api::Type::Enum
                   name: 'signatureAlgorithm'
                   description: |
-                    The algorithm used.
-                  required: true
+                    A signature algorithm that must be used. If this is omitted, any EC-based signature algorithm will be allowed.
```
Comment (roaks3) [inline mmv1/products/privateca/CaPool.yaml:133]: In this case, the `ellipticCurve` object can only have one possible field `signatureAlgorithm`, which I believe is why it is required here. Sometimes empty objects can be used, but it is rare. Would `elliptic_curve = {}` be valid configuration?

My guess is that `signatureAlgorithm` should remain required.

I noticed a similar pattern on some of the other fields as well, and wanted to make sure to clarify: the `required: true` means the field is required within the context of its parent. So for the other nested fields where you've removed `required: true`, I'd like to make sure that the parent object can still be used without those fields.

**Review by roaks3 — CHANGES_REQUESTED** [review]
After looking through, I think some of these `required: true` annotations should remain, but wanted to verify the behavior with you.

Once that is resolved, it still appears that a lot of these would not be covered by tests.

File: mmv1/products/privateca/CaPool.yaml (hunk 1 above)
Comment (gfxcc) [reply mmv1/products/privateca/CaPool.yaml:133]: Thanks for your detailed review. 

Technically, yes. `elliptic_curve = {}` is equivalent to `elliptic_curve = { UNSPECIFIED_ALGO }`, which means [any EC-based signature algorithm will be allowed](http://google3/google/cloud/security/privateca/v1/resources.proto;l=518-519;rcl=549523805). However, it seems the [proto validator](http://google3/google/cloud/security/privateca/v1/resources.proto;l=523;rcl=549523805) blocked that use case. I will work with the team to unblock it.

Similar changes for cl/553167307 .

I will update the PR once it is ready to move forward.

**Pull request closed without merging**